packages/dx-react-scheduler-material-ui/src/templates/date-navigator/calendar/root.jsx

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import moment from 'moment';
+import { Table } from './table';
+
+export class Root extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    const { currentDate, onNavigate } = this.props;
+    this.state = {
+      currentDate,
+    };
+    this.onNavigate = ({ back }) => {
+      const { currentDate: currentDateState } = this.state;
+      const nextDate = moment(currentDateState)[back ? 'subtract' : 'add'](1, 'month');
+      this.setState({ currentDate: nextDate.toDate() });
+    };
+    this.onCellClick = ({ nextDate }) => {
+      onNavigate({ nextDate });
+      this.setState({ currentDate: nextDate });
+    };
+  }
+
+  render() {
+    const {
+      currentDate, firstDayOfWeek, getHeaderCells, getCells,
+      titleComponent: Title,
+      navigationButtonComponent: NavigationButton,
+      navigatorComponent: Navigator,
+      rowComponent: Row,
+      cellComponent: Cell,
+      headerRowComponent: HeaderRow,
+      headerCellComponent: HeaderCell,
+      onNavigate,
+      ...restProps
+    } = this.props;
+    const { currentDate: currentDateState } = this.state;
+    return (
+      <div
+        {...restProps}
+      >
+        <Navigator
+          currentDate={currentDateState}
+          titleComponent={Title}
+          navigationButtonComponent={NavigationButton}
+          onNavigate={this.onNavigate}
+        />
+        <Table
+          headerCells={getHeaderCells(currentDateState, firstDayOfWeek)}
+          cells={getCells(currentDateState, firstDayOfWeek)}
+          rowComponent={Row}
+          cellComponent={Cell}
+          headerRowComponent={HeaderRow}
+          headerCellComponent={HeaderCell}
+          onCellClick={this.onCellClick}
+        />
+      </div>
+    );
+  }
+}
+
+Root.propTypes = {
+  titleComponent: PropTypes.func.isRequired,
+  navigationButtonComponent: PropTypes.func.isRequired,
+  getHeaderCells: PropTypes.func.isRequired,
+  getCells: PropTypes.func.isRequired,
+  rowComponent: PropTypes.func.isRequired,
+  cellComponent: PropTypes.func.isRequired,
+  headerRowComponent: PropTypes.func.isRequired,
+  headerCellComponent: PropTypes.func.isRequired,
+  navigatorComponent: PropTypes.func.isRequired,
+  currentDate: PropTypes.oneOfType([PropTypes.string, PropTypes.instanceOf(Date)]),
+  firstDayOfWeek: PropTypes.number,
+  onNavigate: PropTypes.func,
+};
+
+Root.defaultProps = {
+  currentDate: new Date(),
+  firstDayOfWeek: 1,
```

Comment:
We already define these properties. Maybe make they `required` and remove from `defaultProps`?